acceptance/tests/loader/autoload_from_resource_type_decl.rb

```diff
@@ -0,0 +1,84 @@
+# Verify Resource[]{} syntax works correctly with the autoloader
+# and also works correctly with the loader and generated types
+test_name 'C100303: Resource type statement works with auto-loader and loader with generated types' do
+  tag 'risk:med'
+
+  require 'puppet/acceptance/environment_utils.rb'
+  extend Puppet::Acceptance::EnvironmentUtils
+
+  app_type               = File.basename(__FILE__, '.*')
+  tmp_environment        = mk_tmp_environment_with_teardown(master, app_type)
+  fq_tmp_environmentpath = "#{environmentpath}/#{tmp_environment}"
+
+  relative_type_dir    = "modules/one/lib/puppet/type"
+  relative_type_1_path = "#{relative_type_dir}/type_tst1.rb"
+  relative_type_2_path = "#{relative_type_dir}/type_tst2.rb"
+  step 'create custom type in two environments' do
+    on(master, "mkdir -p #{fq_tmp_environmentpath}/#{relative_type_dir}")
+
+    custom_type_1 = <<-END
+    Puppet::Type.newtype(:type_tst1) do
+      newparam(:name, :namevar => true) do
+        p "found_type_tst1"
+      end
+    end
+    END
+    custom_type_2 = <<-END
```

Comment:
Using two types doesn't really add anything to the test. I think that might be a remnant from my original test that used an `each` call to build statements. That's not needed here.